lib/internal/Magento/Framework/Mail/TransportInterfaceFactory.php

```diff
@@ -49,4 +49,14 @@ public function create(array $data = [])
     {
         return $this->_objectManager->create($this->_instanceName, $data);
     }
+
+    /**
+     * @param $name
+     * @return $this
+     */
+    public function setInstanceName($name)
```

Comment:
I am surprised this factory exists in the codebase as I expect all these to be code generated; I suspect this is a bug will get confirmation from @antonkril -- regardless you should specify a di preference to your own factory in TransportBuilder and not requiring this to be done via a plugin.